### PR TITLE
Templated variables side panel

### PR DESCRIPTION
**Pull request by CharukaK**
## Purpose
This PR implements a side panel to the editor which allows the user to set values for templated variables in Siddhi apps

Following are screenshots of the newly added side panel
Screen Shot 2019-09-26 at 10 46 33 PM
Screen Shot 2019-09-26 at 10 46 53 PM
Screen Shot 2019-09-26 at 10 55 53 PM

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes
 - Ran FindSecurityBugs plugin and verified report? yes
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes

### CODE REVIEW

**Comment by pcnfernando** [comment]
Can you please update the following.
- Replace X icon to delete with the recycle bin which is being used in other places
- Instead of having ClearAll button change it as Delete All image

**Comment by CharukaK** [comment]
Completed the requested changes, please find the UI fixes in the screenshot as shown below 
Screen Shot 2019-09-27 at 10 17 08 AM

File: tooling/components/io.siddhi.distribution.editor.core/src/main/resources/web/index.html [hunk 1]
```diff
@@ -2233,13 +2260,85 @@ <h4 class="modal-title file-dialog-title" id="sampleEvent">Generate Sample Event
...
+<script id="template-deploy-side-panel-template" type="text/x-handlebars-template">
+    <div id="template-deploy" role="tabpanel" class="ws-tab-pane template-deploy">
+        <div class="content">
+            <h3>Templated Variables</h3>
+
+            <form role="form" class="single-event-form" data-form-type="single" data-toggle="loading"
+                  data-loading-style="overlay" data-loading-text="Sending event">
+                <div class="form-group">
+                    <button type="button" class="btn btn-primary"  id="refreshAttributes"
+                        style="right: 10px; top: 10px; position: absolute"
```
Comment (niveathika) [inline tooling/components/io.siddhi.distribution.editor.core/src/main/resources/web/index.html]: Moe styles to css

File: tooling/components/io.siddhi.distribution.editor.core/src/main/resources/web/js/template-deploy/template-deploy.js [hunk 2]
```diff
@@ -0,0 +1,273 @@
...
+                        alerts.error(msg);
+                    }
+                });
+            });
+
+            templateAttrList.sort();
+
+            var localStorageTemplatedAttributes = JSON.parse(localStorage.getItem('templatedAttributeList'));
+
+            console.log("localStorage", _.differenceWith(templateAttrList,
```
Comment (niveathika) [inline tooling/components/io.siddhi.distribution.editor.core/src/main/resources/web/js/template-deploy/template-deploy.js]: Remove console log

File: tooling/components/io.siddhi.distribution.editor.core/src/main/resources/web/js/template-deploy/template-deploy.js [hunk 3]
```diff
@@ -0,0 +1,273 @@
...
+
+                    $("#" + justGetName(attr) + "-input").on('keyup', function (e) {
+                        let attrLocalStore = JSON.parse(localStorage.getItem('templatedAttributeList'));
+                        attrLocalStore['${' + e.target.id.split('-input')[0] + '}'] = e.target.value;
+
+                        localStorage.setItem("templatedAttributeList", JSON.stringify(attrLocalStore));
+                    });
+
+                    $("#" + justGetName(attr) + "-clear").on('click', function (e) {
+                        console.log(e);
```
Comment (niveathika) [inline tooling/components/io.siddhi.distribution.editor.core/src/main/resources/web/js/template-deploy/template-deploy.js]: Remove log In all place

File: tooling/components/io.siddhi.distribution.editor.core/src/main/resources/web/js/template-deploy/template-deploy.js [hunk 4]
```diff
@@ -0,0 +1,273 @@
...
+                JSON.parse(str);
+            } catch (e) {
+                return false;
+            }
+            return true;
+        }
+
+
+        TemplateDeploy.prototype.handleErrorMsg = function (msg) {
+            // todo: check
```
Comment (niveathika) [inline tooling/components/io.siddhi.distribution.editor.core/src/main/resources/web/js/template-deploy/template-deploy.js]: Handle todo

**Pull request merged**